websites/C/Crunchy-DL/presence.ts

```diff
@@ -0,0 +1,90 @@
+const presence = new Presence({
+  clientId: "811867948721504268"
+});
+
+function handlePresenceData(presenceData: PresenceData) {
+  const viewingRoutes = ["/", "/perfil", "/categoria"],
+        path = location.pathname,
+        capitalize = (str: string) => {
+          // lols below.. ignore it
+          return str.split(" ").map((e) => {
+            return e.split("").map((ee, ii) => {
+                return ii == 0 ? ee.toUpperCase() : ee.toLowerCase();
+            }).join("");
+          }).join(" ");
+        };
+
+  if(viewingRoutes.includes("/" + path.split("/")[1])) {
+    presenceData.details = "Visualizando:";
+  }
+
+  if(path == "/") {
+
+    presenceData.state = "Página Inicial";
+
+  } else if(path.includes("/categoria")) {
+    
+    const category = capitalize(document.querySelector(".category > h1").textContent);
+    presenceData.state = category;
+
+  } else if(path.includes("/perfil")) {
+
+    if(path == "/perfil/") {
+      presenceData.state = "Meu Perfil";
+    } else {
+      const activeSubmenu = document.querySelector(".profile-menu li.active") as HTMLElement;
+      presenceData.details = "Perfil:" + document.querySelector(".profile-username").textContent;
+      presenceData.state = activeSubmenu.innerText.trim();
+    }
+
+  } else if(path.includes("/anime") && !path.includes("/colecao")) {
+
+    const prependFavorite = document.querySelector(".fa-star").classList.contains("active") ? "⭐" : "";
+    presenceData.details = "Visualizando Anime " + prependFavorite;
+    presenceData.state = document.querySelector(".anime-name").textContent;
+
+  } else if(path.includes("/episodio")) {
+    
+    try {
+      const getTime = (str: string) => {
+        const parts = str.split(":").map(e => parseInt(e));
+        return parts[0] * 60 + parts[1];
+      },
```

Comment:
also I guess, you could use our function from Presence class
![image](https://user-images.githubusercontent.com/9348108/108495470-1f979780-72a9-11eb-85e8-dfc83d095ff1.png)

Reply:
^